src/EditorFeatures/Core/Implementation/EditAndContinue/ActiveStatementViewTaggerProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EditAndContinue;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue
+{
+    [Export(typeof(IViewTaggerProvider))]
+    [TagType(typeof(ITextMarkerTag))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    internal partial class ActiveStatementViewTaggerProvider : AsynchronousViewTaggerProvider<ITextMarkerTag>
+    {
+        // We want to track text changes so that we can try to only reclassify a method body if
+        // all edits were contained within one.
+        protected override TaggerTextChangeBehavior TextChangeBehavior => TaggerTextChangeBehavior.TrackTextChanges;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ActiveStatementViewTaggerProvider(
+            IThreadingContext threadingContext,
+            IForegroundNotificationService notificationService,
+            IAsynchronousOperationListenerProvider listenerProvider)
+            : base(threadingContext, listenerProvider.GetListener(FeatureAttribute.Classification), notificationService)
+        {
+        }
+
+        protected override ITaggerEventSource CreateEventSource(ITextView textView, ITextBuffer subjectBuffer)
+        {
+            AssertIsForeground();
+            const TaggerDelay Delay = TaggerDelay.Short;
+
+            return TaggerEventSources.Compose(
+                new EventSource(subjectBuffer, Delay),
+                TaggerEventSources.OnViewSpanChanged(ThreadingContext, textView, textChangeDelay: Delay, scrollChangeDelay: TaggerDelay.NearImmediate),
```

Comment:
you're restarting it the view span changes?  that seems odd.  i would have expected this to be a buffer tagger, not a view tagger.

view taggers are useful for when tehre's a *ton* to do and you really want to scope the work down to just what's in view (i.e. semantic classification).   buffer taggers are normally used for almost everything else.

Reply:
I see, I just copied semantic tagger provider :)

Should this type derive from `AbstractAsynchronousTaggerProvider` then?

Reply:
Yup!